src/methods/environment.js

```diff
@@ -77,7 +77,9 @@ export class EnvironmentImplementation {
     // void | ObjectValue | BooleanValue | StringValue | SymbolValue | NumberValue | EnvironmentRecord | AbstractValue;
     if (!base || base instanceof EnvironmentRecord) return false;
     let type = base.getType();
-    return type === BooleanValue || type === StringValue || type === SymbolValue || type === NumberValue;
+    return (
+      type === Value || type === BooleanValue || type === StringValue || type === SymbolValue || type === NumberValue
```

Comment:
Given the semantics specified in the specification, this change seems undesirable.